packages/devtools_app/lib/src/standalone_ui/vs_code/debug_sessions.dart

```diff
@@ -29,88 +30,109 @@ class DebugSessions extends StatelessWidget {
           const Text('Begin a debug session to use DevTools.')
         else
           Table(
-            columnWidths: const {0: FlexColumnWidth()},
-            defaultColumnWidth:
-                FixedColumnWidth(defaultIconSize + defaultSpacing),
+            columnWidths: const {
+              0: FlexColumnWidth(2),
+              // TODO(dantup): Fixed width icons+menu?
+              1: FlexColumnWidth(),
+            },
             defaultVerticalAlignment: TableCellVerticalAlignment.middle,
             children: [
               for (final session in sessions)
-                _createSessionRow(context, session),
+                TableRow(
+                  children: [
+                    Text(
+                      '${session.name} (${session.flutterMode})',
+                      style: Theme.of(context).textTheme.titleSmall,
+                    ),
+                    if (api.capabilities.openDevToolsPage)
+                      _DevToolsMenu(api: api, session: session),
+                  ],
+                ),
             ],
           ),
       ],
     );
   }
+}
+
+class _DevToolsMenu extends StatelessWidget {
+  const _DevToolsMenu({required this.api, required this.session});
+
+  final VsCodeApi api;
+  final VsCodeDebugSession session;
 
-  TableRow _createSessionRow(BuildContext context, VsCodeDebugSession session) {
+  @override
+  Widget build(BuildContext context) {
     // TODO(dantup): What to show if mode is unknown (null)?
-    final name = session.name;
     final mode = session.flutterMode;
     final isDebug = mode == 'debug';
     final isProfile = mode == 'profile';
     // final isRelease = mode == 'release' || mode == 'jit_release';
     final isFlutter = session.debuggerType?.contains('Flutter') ?? false;
 
-    return TableRow(
+    return MenuBar(
       children: [
-        Text(
-          '$name ($mode)',
-          style: Theme.of(context).textTheme.titleSmall,
+        IconButton(
+          onPressed: api.capabilities.hotReload && (isDebug || !isFlutter)
```

Comment:
is `(isDebug || !isFlutter)` the same condition used in VS code? I'm wondering if this would be incorrect for dart web apps, or Dart CLI apps?

Reply:
It is currently what Dart-Code does, yep:

https://github.com/Dart-Code/Dart-Code/blob/3efd2a89bc85a006f7771d92297ebbac93380af2/package.json#L681

I'm not sure about Web, but for Dart CLI there's not profile/release mode so it'll be enabled because of isDebug (my understanding is that `reloadSources` is always available).

If the rules are more complex than this, we should probably find a way for the editors to find out via the VM Service/DDS to avoid putting complex rules into editors (because that gets complicated when they change, since they have to support multiple SDK versions).